sql/2021/performance/tbt.sql

```diff
@@ -0,0 +1,31 @@
+#standardSQL
+# TBT data by threshold
+
+CREATE TEMP FUNCTION getTbtThreshold (lighthouseTbtScore FLOAT64) RETURNS STRING LANGUAGE js AS '''
+  if (lighthouseTbtScore >= 0.75) {
+      return 'GOOD'
+  } else if (lighthouseTbtScore >= 0.25) {
+      return 'NI'
+  } else if (lighthouseTbtScore >= 0) {
+      return 'POOR'
+  } else {
+      return 'NA'
+  }
+''';
```

Comment:
Thanks for checking! I'll adjust the limits though that will make it look even worse. Here is how I did the chart:

![image](https://user-images.githubusercontent.com/4777393/139863292-f781f33e-418f-4269-ad9b-8cbd0675544a.png)

So instead of giving the "score" in the legend, I put in the ms numbers to make it clearer to the audience, like the CWV sections (will adjust the < to <=). I can run the query with both just to confirm the numbers match up.

Reply:
Erm, I got different results for score vs ms:

Using score |   |  
-- | -- | --
Row | threshold | pages
1 | POOR | 7565
2 | NI | 1064
3 | GOOD | 1064
  |   | 9693
Using ms |   |  
Row | thresholdMs | pages
1 | POOR | 5664
2 | GOOD | 2971
3 | NI | 1058
  |   | 9693

Query: 

```sql
#standardSQL
# TBT data by threshold

CREATE TEMP FUNCTION getTbtThreshold (lighthouseTbtScore FLOAT64) RETURNS STRING LANGUAGE js AS '''
  if (lighthouseTbtScore >= 0.90) {
      return 'GOOD'
  } else if (lighthouseTbtScore >= 0.50) {
      return 'NI'
  } else if (lighthouseTbtScore >= 0) {
      return 'POOR'
  } else {
      return 'NA'
  }
''';

CREATE TEMP FUNCTION getTbtThresholdMs (lighthouseTbt FLOAT64) RETURNS STRING LANGUAGE js AS '''
  if (lighthouseTbt <= 200) {
      return 'GOOD'
  } else if (lighthouseTbt <= 600) {
      return 'NI'
  } else if (lighthouseTbt > 600) {
      return 'POOR'
  } else {
      return 'NA'
  }
''';

WITH
tbt_stats AS (
  SELECT
    url,
    getTbtThresholdMs(CAST(JSON_EXTRACT_SCALAR(report, "$.audits.total-blocking-time.numericValue") AS FLOAT64)) AS thresholdMs,
    getTbtThreshold(CAST(JSON_EXTRACT_SCALAR(report, "$.audits.total-blocking-time.score") AS FLOAT64)) AS threshold
  FROM
    --`httparchive.lighthouse.2021_07_01_mobile`
    `httparchive.sample_data.lighthouse_mobile_10k`
)

SELECT
  --threshold,
  thresholdMs,
  COUNT(DISTINCT url) AS pages,
FROM
  tbt_stats
GROUP BY
  thresholdMs
```

Here are the results with the previous scores using the original thresholds:

Using score with 0.75 and 0.25 thresholds |   |  
-- | -- | --
Row | threshold | pages
1 | POOR | 6595
2 | GOOD | 1460
3 | NI | 1638
  |   | 9693

Reply:
(also I just realized that chart didn't update the legend - will fix)

Reply:
Sorry, it turns out the score and thresholds aren't quite equivalent because the score is rounded to a whole value. Just over 200 and 600 will get rounded up to "GOOD" and "NI", respectively (the thresholds are more like 204.8 -> score of 0.895, rounded to 0.90 and 606.4 -> score of 0.495, rounded to 0.50).

However the score is what's used for color coding in the Lighthouse report, so using `tbtScore` for this PR sounds 👍 to me